asdf/_asdf.py

```diff
@@ -807,7 +809,7 @@ def _open_asdf(
                 msg = "Does not appear to be a ASDF file."
                 raise ValueError(msg) from e
             self._file_format_version = cls._parse_header_line(header_line)
-            self.version = self._file_format_version
+            self.version = self.file_format_version
```

Comment:
This seems incorrect. `self.version` is the asdf standard version yet here it is being set to the `file_format_version` (which is always `1.0.0`). `self.version` is overwritten for probably every file a few lines down (from the undocumented `ASDF_STANDARD` comment). As the ASDF standard does not define the comment or what to do in this case I'm inclined to leave in this behavior for this PR (as it's only tangentially related to deprecating `version_map`).

It is likely an improvement to change this to default to the `default_version` if the file doesn't define a version. However this would be a major change (switching from asdf standard 1.0.0 to 1.5.0) if there is any code that relies on this.

Reply:
Good guess!

I changed this to just default to `1.0.0` if no `#ASDF_STANDARD` comment is found. That way it should hopefully not break for `2.0.0`.

44fe69f9ee1abb3ebf2dc58133e45484a3d1c4ef

There were 2 tests that failed following this change. One was filling in the standard version for `#ASDF` (so should have been producing an error but was not), the other was expecting an unsupported standard version warning when the `#ASDF` version was incorrect. Both of these were updated. I wouldn't be surprised if there are more places where these two things are conflated :-/